libkbfs/interfaces.go

```diff
@@ -1447,3 +1447,177 @@ type RekeyQueue interface {
 	// Waits for all queued rekeys to finish
 	Wait(ctx context.Context) error
 }
+
+// ImmutableBareRootMetadata is a read-only interface to the bare serializeable MD that
+// is signed by the reader or writer.
+type ImmutableBareRootMetadata interface {
```

Comment:
So Immutable isn't quite right, because that would imply that the underlying object never changes, which isn't true (i.e., if the underlying object has settable fields.). So this should really be `ReadOnlyBareRootMetadata`. But I'd prefer just `BareRootMetadata` and have the mutable one be `MutableBareRootMetadata`, since the latter is less common.